src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/GenerateSbom.cs

```diff
@@ -0,0 +1,16 @@
+namespace DotNetProjectFile.Analyzers.MsBuild;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class GenerateSbom() : MsBuildProjectFileAnalyzer(Rule.GenerateSbom)
+{
+    /// <inheritdoc />
+    public override IReadOnlyCollection<ProjectFileType> ApplicableTo => ProjectFileTypes.ProjectFile;
+
+    protected override void Register(ProjectFileAnalysisContext context)
+    {
+        if (context.File.IsPackable() && context.File.Property<GenerateSBOM>()?.Value is not true)
```

Comment:
if `Microsoft.Sbom.Targets` is required, shouldn't we also check for that?

Reply:
Working on that.